File: solutions/go/0001-two-sum.go
package main

func twoSum(nums []int, target int) []int {
	rest := make(map[int]int, 0)

	for i, num := range nums {
		if j, exists := rest[target-num]; exists {
			return []int{j, i}
		}
		rest[num] = i
	}

	return nil
}

File: solutions/php/0001-two-sum.php
<?php
class Solution
{
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum(array $nums, int $target): ?array
    {
        $rest = [];

        foreach ($nums as $i => $num) {
            if (array_key_exists($target - $num, $rest)) {
                return [$i, $rest[$target - $num]];
            }

            $rest[$num] = $i;
        }

        return null;
    }
}
